helper/resource/testing.go

```diff
@@ -476,6 +476,12 @@ type TestStep struct {
 	// resources in the root module path.
 	Taint []string
 
+	// RemoveState list of resource addresses to be removed from state after
+	// applying config. Be sure to only include this at a step where the referenced
+	// address will be present in state, as it will fail the test if the resource
+	// is missing.
+	RemoveState []string
```

Comment:
I think it would make sense to move this definition under the comments mentioning `Test modes` on line 486. As the intention is for this `RemoveState` option to be it's own "mode".

Maybe also adding to this doc comment that `RemoveState` is not meant to be used in the same `TestStep` with other modes like `Config` or `ImportState`?